### PR TITLE
17051 - migrated Checkbox to PressableWithFeedback

### PR DESCRIPTION
**Pull request by Skalakid**
### Fixed Issues

$ https://github.com/Expensify/App/issues/17051 
PROPOSAL: https://github.com/Expensify/App/issues/17051 

### Tests

1. Log in
2. Enter `Settings` > `Payments`
3. Click `Add payment method`
4. Choose `Debit card` 
5. Click the checkbox `I accept the Expensify Terms of Service`, verify styling and behavior

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests
### QA Steps

Same as Tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/39538890/3cc85c1b-4980-4496-9ed0-6abcc75afc8f

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/39538890/607a6023-1a85-49ac-90ba-30aa9632dfea

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/39538890/469f09b4-c578-40a1-930f-bcfcb572ce1d

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/39538890/18b3c1d8-2146-46fc-a493-84b2941bb8a2

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/39538890/89bc48a8-ad5e-4f42-adc8-27d772a66df0

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/39538890/55940105-5113-4e84-995b-2d46b5956f30

</details>

### CODE REVIEW

**Comment by abdulrahuman5196** [comment]
Will be working on C+ review

**Review by abdulrahuman5196 — COMMENTED** [review]
I will be working on C+ review

**Comment by abdulrahuman5196** [comment]
@Skalakid Kindly fill the Offline test and QA Steps section. If its same as the test, then add "Same as Tests" in the section.

File: src/components/Checkbox.js
Comment (Skalakid) [reply src/components/Checkbox.js:52]: Good point, I will change that

File: src/components/Checkbox.js
Comment (Skalakid) [reply src/components/Checkbox.js:89]: Previously after double-clicking the checkbox icon was somehow selectable, now after I merged the latest main I can't reproduce it so I can remove it

File: src/components/Checkbox.js
Comment (abdulrahuman5196) [inline src/components/Checkbox.js:96]: can we remove this? Since currently the default for hoverDimmingValue is 1?

**Review by abdulrahuman5196 — CHANGES_REQUESTED** [review]

File: src/components/Checkbox.js
Comment (abdulrahuman5196) [inline src/components/Checkbox.js:88]: @Skalakid What is this `this.onBlur`? I always get undefined value for this?

Comment (abdulrahuman5196) [reply src/components/Checkbox.js:88]: @Skalakid  any update on this?

Comment (Skalakid) [reply src/components/Checkbox.js]: @abdulrahuman5196 Hey, sorry for the late response, I just fixed this by merging with the main ;) I think it was added after merging with main, now it's gone

**Comment by abdulrahuman5196** [comment]
@Skalakid The eye icon on the password entry field of pdf attachment is moved up due to this change.
Note: The eye also uses the Checkbox component.

Steps
1) Add password protected pdf to a chat
2) choose enter password
3) eye icon should be middle 

**main branch**

Screenshot 2023-06-20 at 9 44 10 PM


**this PR with PressableWithFeedback**

Screenshot 2023-06-20 at 9 38 24 PM

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>




https://github.com/Expensify/App/assets/46707890/0da9354e-e7d2-4e33-9c2b-f9955e5addb6

</details>

<details>
<summary>Mobile Web - Chrome</summary>





https://github.com/Expensify/App/assets/46707890/35ce4330-16cb-4b1d-9592-37bb511a6138


</details>

<details>
<summary>Mobile Web - Safari</summary>






https://github.com/Expensify/App/assets/46707890/3ca5d47b-86aa-484e-afd0-92cc8ba2c2f7




</details>

<details>
<summary>Desktop</summary>






https://github.com/Expensify/App/assets/46707890/a938a7bf-09c3-450a-bcc7-84cc478c81f8



</details>

<details>
<summary>iOS</summary>





https://github.com/Expensify/App/assets/46707890/7b979413-68d3-471e-bd88-2f3a1fb24a8c


</details>

<details>
<summary>Android</summary>





https://github.com/Expensify/App/assets/46707890/f256c6f5-75e6-467a-b800-8faab69ea998




</details>

File: src/components/Checkbox.js
Comment (abdulrahuman5196) [inline src/components/Checkbox.js:39]: @Skalakid  Wrong type comment here. Could you update this?

**Review by abdulrahuman5196 — CHANGES_REQUESTED** [review]

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours @marcaaron 

🎀👀🎀
C+ Reviewed

File: src/components/TextInput/BaseTextInput.js
Comment (roryabraham) [inline src/components/TextInput/BaseTextInput.js:345]: Don't we need to pass `accessibilityLabel` here?

File: src/components/Checkbox.js
Comment (roryabraham) [inline src/components/Checkbox.js:52]: I don't think it makes sense to have the `accessibilityLabel` default to an empty string. Let's make it required.

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: src/components/TextInput/BaseTextInput.js
Comment (Skalakid) [reply src/components/TextInput/BaseTextInput.js:346]: Yes, I just add it here and in other components

**Comment by marcaaron** [comment]
Gonna unassign myself from reviews here as @roryabraham is managing this project and I think as the assignee to the original issue is the "correct reviewer". That said, if you need an extra set of eyes let me know and I'll be happy to jump back in!

**Comment by roryabraham** [comment]
@Skalakid Can you resolve conflicts here please?

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
Not an emergency, was just too trigger-happy with the merge button and my approval kicked off a new check for the checklist

File: src/components/Checkbox.js
Comment (eVoloshchak) [inline src/components/Checkbox.js:75]: 👋 Just a heads-up that this has caused a small regression in https://github.com/Expensify/App/issues/22413
`BaseGenericPressable` didn't have the logic to disable `onKeyDown` handler when `disabled` prop is true
We resolved this by adding the following condition
```js
onKeyDown={!isDisabled ? onKeyDown : undefined}

**Pull request merged**
